common/spec/specfiles.go

```diff
@@ -206,6 +212,12 @@ func ValidateSpec(files []File, isTargetMandatory, isSearchBasedSpec bool) error
 		if isSearchBasedSpec && !isAql && !isPattern && !isBuild && !isBundle {
 			return errorutils.CheckErrorf("spec must include either aql, pattern, build or bundle")
 		}
+		if isPathMapping && isTarget {
+			return fileSpecValidationError("mappings", "target")
+		}
+		if isPathMapping && isPattern {
+			return fileSpecValidationError("mappings", "pattern")
+		}
```

Comment:
Let's place the above line right after the following code for better readability - 
```
if isPathMapping && !isAql {
	return errorutils.CheckErrorf("mappings supported only with Aql")
}
```